src/lib/InstantSearch.js

```diff
@@ -109,14 +120,25 @@ Usage: instantsearch({
       this.searchParameters
     );
 
+    if (this._searchFunction) {
+      this._originalHelperSearch = helper.search.bind(helper);
+      helper.search = this._wrappedSearch.bind(this);
```

Comment:
But should be ok, since this helper belongs to instantsearch instance, its not created by the user and then injected in the lib.

We do expose instantsearch.helper but its pretty clear thats the one from the helper, and it this case .search will behave differently.